### PR TITLE
Nodejs project

### PR DESCRIPTION
**Pull request by cancerberoSgx**
I think this is done. The only missing piece is you to run "npm publish"

After that we can adjust / add the information on how to install the library with npm in the README.md. I didn't wanted to do that yet  until npm publish. 

I didn't include any typescript bindings or typescript configuration yet. typescript now is only needed as a compile tool - there is no .ts or .d.ts files yet. 

After this is merged I can work on that since I already have the bindings in my TS projects

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "description": "Webassembly compilation of ImageMagick",
+  "main": "dist/magickApi.js",
+  "scripts": {
+    "test": "echo \"No tests :(\" && exit 1",
+    "postinstall": "npm i --no-save download-github-release && npx download-github-release knicknic wasm-imagemagick",
+    "prepare": "tsc --skipLibCheck --allowJs magickApi.js --outDir dist"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cancerberosgx/WASM-ImageMagick.git"
```
Comment (KnicKnic) [inline package.json]: should all url's be manually changed to https://github.com/KnicKnic/WASM-ImageMagick.git or will publishing take care of that?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "webassembly",
+    "imagemagick",
+    "browser",
+    "image",
+    "processing"
+  ],
+  "author": "KnicKnic",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/cancerberosgx/WASM-ImageMagick/issues"
```
Comment (KnicKnic) [inline package.json]: url again

File: package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  },
+  "homepage": "https://github.com/cancerberosgx/WASM-ImageMagick#readme",
```
Comment (KnicKnic) [inline package.json]: url

**Comment by KnicKnic** [comment]
Thank you so much for doing this. I still have yet to figure out npm. I cloned your branch and coppied the realease files to dist.. (not sure if I was supposed to do that) and ran npm publish. I leave this pr open for you to continue to update.

https://www.npmjs.com/package/wasm-imagemagick

**Comment by cancerberoSgx** [comment]
I think this is ready. Using it online at https://codesandbox.io/s/lp7lxz6l59 (placed a link to this playground in the readme because I think will be of help to newcomers). I'm using your npm published packaged in that project and works fine. I needed only to manually copy magick.* to the folder and that was all. So that's all for me in this PR

**Comment by cancerberoSgx** [comment]
> Thank you so much for doing this. I still have yet to figure out npm. I cloned your branch and coppied the realease files to dist.. (not sure if I was supposed to do that) and ran npm publish. I leave this pr open for you to continue to update.

that was not the idea. Right now when users perform "npm install wasm-imagemagick" the magick.* files are downloaded automatically to node_modules/wasm-imagemagic folder so there's no need for you to copy those to "dist" before "npm publish". Just "npm publish" will suffice in the future. 

Otherwhise, if we want to copy the files as you did before publish, that could also be a great idea, so I can move the "downloading" bits to "npm run prepare" that runs just before "npm publish" so npm do the job for you. 

Don't know if you follow me on this one and if so which way do you prefer (store magick.* files in npmjs.org as you did or user's download them when the perform "npm install wasm-imagemagick" automatically ?

**Comment by KnicKnic** [comment]
Thx I completed work item https://github.com/KnicKnic/WASM-ImageMagick/projects/1#card-13115278

**Comment by KnicKnic** [comment]
> > Thank you so much for doing this. I still have yet to figure out npm. I cloned your branch and coppied the realease files to dist.. (not sure if I was supposed to do that) and ran npm publish. I leave this pr open for you to continue to update.
> 
> that was not the idea. Right now when users perform "npm install wasm-imagemagick" the magick.* files are downloaded automatically to node_modules/wasm-imagemagic folder so there's no need for you to copy those to "dist" before "npm publish". Just "npm publish" will suffice in the future.
> 
> Otherwhise, if we want to copy the files as you did before publish, that could also be a great idea, so I can move the "downloading" bits to "npm run prepare" that runs just before "npm publish" so npm do the job for you.
> 
> Don't know if you follow me on this one and if so which way do you prefer (store magick.* files in npmjs.org as you did or user's download them when the perform "npm install wasm-imagemagick" automatically ?

I did not notice your response when I closed the PR.

I would prefer to not checkin binaries(wasm file) into github (This is what I thought you meant in email when you were talking about checking in binaries, but now I realize it was check them into npm, sorry). Also I would like for the npm package to be fully versioned (if a user selects an older version, they get the older wasm).

Whether npm downloads them from github release or not I do not have the strongest conviction (npm is new to me), but would lean towards putting the binaries & javascript objects into npm.
Reasoning being is the package can be more easily cached and checked into offline repositories. As well as compatibility with any other tooling that deals with npm packages.

**Comment by cancerberoSgx** [comment]
> but would lean towards putting the binaries & javascript objects into npm.
> Reasoning being is the package can be more easily cached and checked into offline repositories. As well as compatibility with any other tooling that deals with npm packages.

@KnicKnic I agree, Will make the changes in another PR.

**Pull request merged**